tests/test_core.py

```diff
@@ -914,6 +914,29 @@ def test_emmalloc(self, *args):
                 open(path_from_root('tests', 'core', 'test_emmalloc.cpp')).read(),
                 open(path_from_root('tests', 'core', 'test_emmalloc.txt')).read())
 
+  def test_emmalloc_usable_size(self, *args):
+    self.set_setting('MALLOC', 'emmalloc')
+    self.emcc_args += list(args)
+
+    self.do_run(open(path_from_root('tests', 'test_malloc_usable_size.c')).read(),
+                open(path_from_root('tests', 'test_malloc_usable_size.txt')).read())
+
+  @no_fastcomp('this feature works in fastcomp, but test outputs are sensitive to wasm backend')
+  def test_emmalloc_memory_statistics(self, *args):
+    self.set_setting('MALLOC', 'emmalloc')
+    self.emcc_args += ['-s', 'TOTAL_MEMORY=128MB', '-g'] + list(args)
+
+    self.do_run(open(path_from_root('tests', 'test_emmalloc_memory_statistics.cpp')).read(),
+                open(path_from_root('tests', 'test_emmalloc_memory_statistics.txt')).read())
+
```

Comment:
We have `do_run_in_out_file_test` for this pattern.

Reply:
check